File: src/main/java/de/maggiwuerze/xdccloader/controller/UserController.java
package de.maggiwuerze.xdccloader.controller;

import de.maggiwuerze.xdccloader.model.entity.UserSettings;
import de.maggiwuerze.xdccloader.model.forms.UserSettingsForm;
import de.maggiwuerze.xdccloader.service.UserSettingsService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;


@Controller
@Slf4j
@RequiredArgsConstructor
class UserController {

	private final UserSettingsService userSettingsService;

	@PostMapping(value = "/usersettings")
	public ResponseEntity<?> updateUserSettings(
		@RequestBody UserSettingsForm userSettingsForm) {
		UserSettings userSettingsById = userSettingsService.getUserSettings();
		userSettingsById.setDownloadSortBy(userSettingsForm.getDownloadSortBy());
		userSettingsById.setSessionTimeout(userSettingsForm.getSessionTimeout());
		userSettingsService.saveUserSettings(userSettingsById);

		return new ResponseEntity<>("UserSettings updated successfully.", HttpStatus.OK);
	}
}

File: src/main/kotlin/de/maggiwuerze/xdccwebloader/controller/UserController.kt
package de.maggiwuerze.xdccwebloader.controller

import de.maggiwuerze.xdccwebloader.persistence.entity.UserSettings
import de.maggiwuerze.xdccwebloader.persistence.entity.UserSettingsTO
import de.maggiwuerze.xdccwebloader.model.forms.UserSettingsFormTO
import de.maggiwuerze.xdccwebloader.service.UserSettingsService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping

@Controller
@RequestMapping("api/v1//")
class UserController(val userSettingsService: UserSettingsService) {

    @GetMapping(value = ["usersettings"])
    fun getSetting(): ResponseEntity<UserSettingsTO> =
        ResponseEntity(userSettingsService.getUserSettings().toTO(), HttpStatus.OK)


    @PostMapping(value = ["usersettings"])
    fun updateSetting(
        @RequestBody userSettingsFormTO: UserSettingsFormTO
    ): ResponseEntity<*> {
        val userSettingsById: UserSettings = userSettingsService.getUserSettings()
        userSettingsById.downloadSortBy = userSettingsFormTO.downloadSortBy
        userSettingsById.sessionTimeout = userSettingsFormTO.sessionTimeout
        userSettingsService.saveUserSettings(userSettingsById)

        return ResponseEntity("UserSettings updated successfully.", HttpStatus.OK)
    }
}